packages/instantsearch-react/src/connectors/connectHierarchicalMenu.js

```diff
@@ -116,7 +116,7 @@ export default createConnector({
     const selectedItem = getSelectedItem(props, state);
     return {
       id,
-      filters: selectedItem === null ? [] : [{
+      filters: !selectedItem ? [] : [{
```

Comment:
https://github.com/algolia/instantsearch.js/pull/1188/commits/0609244f5a883922e4e863e9acba8179e0b6148c

Should fix this, by having selectedItem always be either `null` or a non-empty string. On the other hand, the actual value in state should always be a string, and the empty string represents the absence of a value.